packages/mobile/src/transactions/types.ts

```diff
@@ -26,6 +28,27 @@ export interface TransferStandby {
 
 export type StandbyTransaction = ExchangeStandby | TransferStandby
 
+// Context used for logging the transaction execution flow.
+export interface TransactionContext {
+  // Unique identifier used for tracking a transaction within logging.
+  // Note that this is not the transaction hash, which is unknown when creating a new transaction.
+  id: string
```

Comment:
Longer term could consider precomputing the transaction hash? If all fields are specified before we start tracking

Reply:
In cases where we really do know all the transaction fields on the outset, this would definitely be a nice way to handle tracking. It's likely that would be a rare case though because fields like `nonce` and `gas` require a chain state (or at least a model of chain state) to set correctly, so we currently rely on the node to fill them. It would also break our current abstraction layers to calculate the hash before signing, as that's delegated to the wallet interface. It's important that we can link the transaction hash and this local identifier, and the current method is that in the logs, and in the analytics events, the hash is included, alongside the ID, when it is received after signing as part of the `transaction_hash_received` event.